### PR TITLE
Update to the TEP workflow

### PR DESCRIPTION
**Pull request by afrittoli**
Update the TEP workflow to make the PR that changes the status
of a proposed TEP the central place to track discussion on the
TEP.

This was discussed during an API WG.

### CODE REVIEW

File: teps/0001-tekton-enhancement-proposal-process.md [hunk 1]
```diff
@@ -283,14 +283,28 @@ A TEP has the following states
...
+
+The discussion on the TEP shall be tracked on the PR, regardless of
+the forum where it happens. We might need more information
 about the impact on users, or some time to socialize it with the
 Working Groups, etc. This state doesn't prevent using
 `tektoncd/experimental` to *experiment* and gather feedback.
 
+The outcome may be that the TEP is approved, and moves to
+`implementable`, or rejected, and moves to `withdrawn`. In case the
+TEP is `witdhdrawn` it's best practise to update it with the reason
```
Comment (ImJasonH) [inline teps/0001-tekton-enhancement-proposal-process.md:305]: typo: withdrawn

File: teps/0001-tekton-enhancement-proposal-process.md [hunk 2]
```diff
@@ -283,14 +283,28 @@ A TEP has the following states
...
+for widthdrwal.
```
Comment (ImJasonH) [inline teps/0001-tekton-enhancement-proposal-process.md:306]: typo: withdrawal

**Comment by bobcatfish** [comment]
/lgtm

I'm assuming enough of us have looked it so I'm gonna throw in:

/approve

Thanks @afrittoli !

**Pull request merged**